scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,882 @@
+    return xi.items.A_EGG + letter
+end
+
+xi.events.egg_hunt.eggInList = function(eggs, egg)
+    for j = 1, #eggs do
+        if eggs[j] and eggs[j][1] == egg then
+            return j
+        end
+    end
+
+    return -1
+end
+
+xi.events.egg_hunt.stringToEggs = function(text)
+    local str = string.lower(text)
+    local eggs = {}
+
+    for i = 1, #str do
+        local letter = string.byte(string.sub(str, i, i))
+        local egg = xi.items.A_EGG + (letter - 97)
+        local pos = xi.events.egg_hunt.eggInList(eggs, egg)
+
+        if pos > -1 then
+            eggs[pos][2] = eggs[pos][2] + 1
+        else
+            table.insert(eggs, { egg, 1 })
+        end
+    end
+
+    return eggs
+end
+
+local hasFirstThree = function(player)
+    return (
+        player:getVar(settings.VAR.FIRST_THREE) == 1 or
+        player:hasItem(xi.items.EGG_HELM)
+    )
+end
+
+-- ----------------------------------
+-- Combos
+-- ----------------------------------
+
+local minorRewards =
+{
+    xi.items.CHOCOBO_TICKET,
+    xi.items.RED_DROP,
+    xi.items.YELLOW_DROP,
+    xi.items.BLUE_DROP,
+    xi.items.GREEN_DROP,
+    xi.items.CLEAR_DROP,
+    xi.items.PURPLE_DROP,
+    xi.items.WHITE_DROP,
+    xi.items.BLACK_DROP,
+}
+
+local minorReward = function()
+    return minorRewards[math.random(#minorRewards)]
+end
+
+local tradeIn = function(player, npc, trade, zoneID)
+    local reward = {}
+
+    if
+        npcUtil.tradeHasExactly(trade, { { xi.items.BIRD_EGG, 12 } }) or
+        npcUtil.tradeHasExactly(trade, { { xi.items.LIZARD_EGG, 6 } }) or
+        npcUtil.tradeHasExactly(trade, xi.items.SOFT_BOILED_EGG) or
+        npcUtil.tradeHasExactly(trade, xi.items.COLORED_EGG) or
+        (settings.ERA_2008 and npcUtil.tradeHasExactly(trade, { { xi.items.HARD_BOILED_EGG, 12 } })) or
+        (settings.ERA_2019 and npcUtil.tradeHasExactly(trade, xi.items.APKALLU_EGG))
+    then
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+    end
+
+    if npcUtil.tradeHasExactly(trade, xi.items.PARTY_EGG) then
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+    end
+
+    if
+        npcUtil.tradeHasExactly(trade, xi.items.LUCKY_EGG) or
+        (settings.ERA_2018 and npcUtil.tradeHasExactly(trade, { { xi.items.SAIRUI_RAN, 99 } }))
+    then
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+        table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+    end
+
+    if
+        settings.ERA_2018 and
+        npcUtil.tradeHasExactly(trade, xi.items.IMPERIAL_EGG)
+    then
+        for i = 1, 8 do
+            table.insert(reward, math.random(xi.items.A_EGG, xi.items.Z_EGG))
+        end
+    end
+
+    if #reward > 0 then
+        if player:getVar(settings.VAR.DAILY_BONUS) >= vanaDay() then
```

Comment:
There are more examples, but my previous comment applies to all cases of getVar and setVar